File: ZipFlash_debugBuild/app/src/main/java/com/zipflash/mrp/helper/CheckPermHelper.java
package com.zipflash.mrp.helper;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.preference.PreferenceManager;

import androidx.core.content.ContextCompat;

/**
 * Helper class for checking Shizuku and WRITE_SECURE_SETTINGS permissions,
 * and managing skip-state logic for limited mode.
 */
public class CheckPermHelper {
	
	// --- Temporary "don't show again" flag (resets when app restarts) ---
	private static boolean hideLimitedWarning = false;

	public static boolean getHideLimitedWarning() {
		return hideLimitedWarning;
	}

	public static void setHideLimitedWarning(boolean hide) {
		hideLimitedWarning = hide;
	}

    // Temporary in-memory skip (resets when app restarts)
    private static boolean skipShizukuTransient = false;

    // Check if Shizuku permission is granted
    public static boolean hasShizukuPermission() {
        try {
            return (rikka.shizuku.Shizuku.getVersion() > 0)
				&& (rikka.shizuku.Shizuku.checkSelfPermission()
				== PackageManager.PERMISSION_GRANTED);
        } catch (Throwable e) {
            return false;
        }
    }

    // Check if WRITE_SECURE_SETTINGS is granted
    public static boolean hasWriteSecureSettings(Context ctx) {
        return ContextCompat.checkSelfPermission(
			ctx, "android.permission.WRITE_SECURE_SETTINGS")
			== PackageManager.PERMISSION_GRANTED;
    }

    // Persistent skip flag ("Don't show again" until app restart)
    public static boolean isSkipShizukuPersisted(Context ctx) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
        return prefs.getBoolean("skip_shizuku", false);
    }

    // Transient skip (resets when app closes)
    public static boolean isSkipShizukuTransient() {
        return skipShizukuTransient;
    }

    // Global method to check if Shizuku should be skipped (persistent or transient)
    public static boolean isSkipShizuku(Context ctx) {
        return isSkipShizukuTransient() || isSkipShizukuPersisted(ctx);
    }

    // Save skip flags
    public static void setSkipShizukuPersisted(Context ctx, boolean value) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
        prefs.edit().putBoolean("skip_shizuku", value).apply();
    }

    public static void setSkipShizukuTransient(boolean value) {
        skipShizukuTransient = value;
    }

    // Reset transient flag (called when app restarts or CheckPerm closes)
    public static void resetSkipFlags(Context ctx) {
        skipShizukuTransient = false;
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
        prefs.edit().remove("skip_shizuku").apply();
    }
}

File: ZipFlash/app/src/main/java/com/zipflash/mrp/helper/CheckPermHelper.kt
package com.zipflash.mrp.helper

import android.content.Context
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat

object CheckPermHelper {
    private var hideLimitedWarning = false
    private var skipShizukuTransient = false

    fun getHideLimitedWarning(): Boolean = hideLimitedWarning

    fun setHideLimitedWarning(hide: Boolean) {
        hideLimitedWarning = hide
    }

    fun hasShizukuPermission(): Boolean {
        return try {
            rikka.shizuku.Shizuku.getVersion() > 0
                && rikka.shizuku.Shizuku.checkSelfPermission() == PackageManager.PERMISSION_GRANTED
        } catch (e: Throwable) {
            false
        }
    }

    fun hasWriteSecureSettings(ctx: Context): Boolean {
        return ContextCompat.checkSelfPermission(ctx, "android.permission.WRITE_SECURE_SETTINGS") == PackageManager.PERMISSION_GRANTED
    }

    fun isSkipShizukuPersisted(ctx: Context): Boolean {
        val prefs = ctx.getSharedPreferences("AppSettings", Context.MODE_PRIVATE)
        return prefs.getBoolean("skip_shizuku", false)
    }

    fun isSkipShizukuTransient(): Boolean = skipShizukuTransient

    fun isSkipShizuku(ctx: Context): Boolean = isSkipShizukuTransient() || isSkipShizukuPersisted(ctx)

    fun setSkipShizukuPersisted(ctx: Context, value: Boolean) {
        val prefs = ctx.getSharedPreferences("AppSettings", Context.MODE_PRIVATE)
        prefs.edit().putBoolean("skip_shizuku", value).apply()
    }

    fun setSkipShizukuTransient(value: Boolean) {
        skipShizukuTransient = value
    }

    fun resetSkipFlags(ctx: Context) {
        skipShizukuTransient = false
        val prefs = ctx.getSharedPreferences("AppSettings", Context.MODE_PRIVATE)
        prefs.edit().remove("skip_shizuku").apply()
    }
}
